artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -109,66 +111,109 @@ async def add_builds(self, builds: typing.List[konflux_build_record.KonfluxBuild
         with concurrent.futures.ThreadPoolExecutor() as pool:
             await asyncio.gather(*(loop.run_in_executor(pool, self.add_build, build) for build in builds))
 
-    async def search_builds_by_fields(self, start_search: datetime, end_search: datetime = None,
-                                      where: typing.Dict[str, typing.Any] = None, extra_patterns: dict = None,
-                                      order_by: str = '', sorting: str = 'DESC', limit: int = None) -> RowIterator:
+    async def search_builds_by_fields(
+            self,
+            start_search: typing.Optional[datetime] = None,
+            end_search: typing.Optional[datetime] = None,
+            window_size: int = DEFAULT_SEARCH_WINDOW,
+            where: typing.Optional[typing.Dict[str, typing.Any]] = None,
+            extra_patterns: typing.Optional[dict] = None,
+            order_by: str = '',
+            sorting: str = 'DESC',
+            limit: typing.Optional[int] = None,
+    ):
         """
         Execute a SELECT * from the BigQuery table.
 
         "where" is an optional dictionary that maps names and values to define a WHERE clause.
-        "start_search" is a lower bound to be applied to the partitioning field `start_time`.
-        "end_search" can optionally be provided as an upper bound for the same field.
-
-        Return a (possibly empty) list of results
+        "start_search" is a lower bound to be applied to the partitioning field `start_time`. If None, the search starts 360 days ago.
+        "end_search" can optionally be provided as an upper bound for the same field. If None, the search ends now.
+        "window_size" is the number of days to search in each iteration. If 0, the whole period is searched at once.
+        "extra_patterns" is an optional dictionary that maps names and values to define extra patterns to be matched.
+        "order_by" is the column to order by.
+        "sorting" is the sorting order.
+        "limit" is the maximum number of results to return. None for no limit.
+
+        Return a generator that yields KonfluxRecord objects.
         """
 
-        where_clauses = [
-            Column('start_time', DateTime) >= start_search,
-        ]
-
-        if end_search:
-            where_clauses.append(Column('start_time', DateTime) < end_search)
-
-        where = where if where else {}
+        if start_search and end_search and start_search >= end_search:
+            raise ValueError(f"start_search {start_search} must be earlier than end_search {end_search}")
+        end_search = end_search or datetime.now(tz=timezone.utc)
+        start_search = start_search or end_search - timedelta(days=DEFAULT_SEARCH_DAYS)
+        if window_size < 0:
```

Comment:
I think this might be clearer if we enforce `window_size` to be > 0, and default it to None
```
    async def search_builds_by_fields(
            ...
            window_size: int = None,
...
        window_size =  window_size if window_size else DEFAULT_SEARCH_WINDOW
        assert window_size > 0, f"search_window {window_size} must be a non-negative integer"
```

Reply:
makes sense